solana/solana-ibc/programs/solana-ibc/src/transfer/impls.rs

```diff
@@ -34,8 +139,41 @@ impl TokenTransferExecutionContext for IbcStorage<'_, '_> {
             amt.denom.trace_path,
             amt.denom.base_denom
         );
+        let receiver_id = account
+            .try_into()
+            .map_err(|_| TokenTransferError::ParseAccountFailure)?;
+        let base_denom = amt.denom.base_denom.to_string();
+        let amount_in_u64 = check_amount_overflow(amt.amount)?;
+
+        let (token_mint_key, _bump) =
+            Pubkey::find_program_address(&[base_denom.as_ref()], &crate::ID);
+        let (mint_authority_key, mint_authority_bump) =
+            Pubkey::find_program_address(&[MINT_ESCROW_SEED], &crate::ID);
+        let store = self.borrow();
+        let accounts = &store.accounts;
+        let receiver = get_account_info_from_key(accounts, receiver_id)?;
+        let token_mint = get_account_info_from_key(accounts, token_mint_key)?;
+        let token_program = get_account_info_from_key(accounts, spl_token::ID)?;
+        let mint_authority =
+            get_account_info_from_key(accounts, mint_authority_key)?;
+
+        let bump_vector = mint_authority_bump.to_le_bytes();
+        let inner = vec![MINT_ESCROW_SEED, bump_vector.as_ref()];
+        let outer = vec![inner.as_slice()];
```

Comment:
No need for vec as before. We probably need a function for this. One which takes reference to `mint_authority_bump` and returns `&[&[&[u8]]]` for the seeds.